core/ir/disassemble_shared.c

```diff
@@ -749,9 +751,15 @@ internal_opnd_disassemble(char *buf, size_t bufsz, size_t *sofar DR_PARAM_INOUT,
         case IMMED_DOUBLE_kind:
         case PC_kind:
         case FAR_PC_kind: break;
-        case REG_kind:
-            if (!opnd_is_reg_partial(opnd))
+        case REG_kind: {
+            /* We always want to print the operation size for DR_ISA_REGDEPS
+             * instructions.
+             */
+            bool is_global_isa_mode_synthetic =
+                dr_get_isa_mode(get_thread_private_dcontext()) == DR_ISA_REGDEPS;
```

Comment:
I thought the mode of a decoded instruction was stored in the instr_t and not in the dcontext.  Maybe the instr_t needs to be passed in here.

Reply:
Removed the need for this check altogether.